native/cocos/bindings/jswrapper/v8/Utils.cpp

```diff
@@ -243,6 +243,39 @@ void clearPrivate(v8::Isolate *isolate, ObjectWrap &wrap) {
     }
 }
 
+// ExternalStringResource
+ExternalStringResource::ExternalStringResource(std::u16string &&s)
+: _s(std::move(s)) {
```

Comment:
s is right reference, should use std::move?

Reply:
Yes, need it.
Test code:

```cpp
class Foo {
public:
    Foo(int a)
    : a(a)
    {
        printf("constructor\n");
    }
    
    ~Foo() {
        printf("~Foo: %d\n", a);
    }
    
    Foo(const Foo& o) {
        a = o.a;
        printf("copy constructor ...\n");
    }
    
    Foo(Foo && o) {
        a = o.a;
        o.a = 0;
        printf("move constructor ...\n");
    }
    
    Foo& operator=(const Foo &o) {
        a = o.a;
        printf("copy assignment ...\n");
        return *this;
    }
    
    Foo& operator=(Foo &&o) {
        a = o.a;
        o.a = 0;
        printf("move assignment ...\n");
        return *this;
    }
    
    int a = 0;
};
```

```cpp
    class Bar {
    public:
        Bar(Foo && foo)
        : _foo(std::move(foo))
        {}
        
    private:
        Foo _foo;
    };
    
    {
        Foo hello(123);
        Bar b(std::move(hello));
        printf("after bar\n");
    }
/*
constructor
move constructor ...
after bar
~Foo: 123
~Foo: 0
*/
```

```cpp
    class Bar {
    public:
        Bar(Foo && foo)
        : _foo(foo)
        {}
        
    private:
        Foo _foo;
    };
    
    {
        Foo hello(123);
        Bar b(std::move(hello));
        printf("after bar\n");
    }

/*
constructor
copy constructor ...
after bar
~Foo: 123
~Foo: 123
*/
```

Reply:
good to know